### PR TITLE
Conditionalize platform tests

### PR DESCRIPTION
**Pull request by sean-**

### CODE REVIEW

File: client/allocdir/alloc_dir_test.go [hunk 1]
```diff
@@ -5,13 +5,18 @@ import (
...
+	"runtime"
 	"testing"
 
 	"github.com/hashicorp/nomad/client/testutil"
 	"github.com/hashicorp/nomad/nomad/structs"
 )
 
 var (
+	osMountSharedDirSupport = map[string]bool{
+		"linux": true,
```
Comment (dadgar) [inline client/allocdir/alloc_dir_test.go]: darwin has the ability as well

File: client/fingerprint/cgroup_default.go [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// +build !linux
+// +build
```
Comment (dadgar) [inline client/fingerprint/cgroup_default.go]: Remove if there is no conditional build?

Comment (sean-) [inline client/fingerprint/cgroup_default.go]: `s/can't/can/`, right?

**Comment by sean-** [comment]
Implicit LGTM from status call: @dadgar

**Pull request merged**